src/components/TodoItem.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useEffect, useRef } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  isLoading: boolean;
+  deleteTodo: (id: number) => void;
+  handleEdit: (todo: Todo) => void;
+  activeTodo: (id: number, completed: boolean) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  isLoading,
+  deleteTodo,
+  handleEdit,
+  activeTodo,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [todoTitle, setTodoTitle] = useState(todo.title);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (isEditing && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isEditing]);
+
+  const handleEditSubmit = async () => {
+    const trimmedTodoTitle = todoTitle.trim();
+
+    if (!trimmedTodoTitle) {
+      deleteTodo(todo.id);
+    } else if (trimmedTodoTitle !== todo.title) {
+      const updatedTodo = { ...todo, title: trimmedTodoTitle };
+
+      setTodoTitle(updatedTodo.title);
+      setIsEditing(false);
+      handleEdit(updatedTodo);
+    } else {
+      setIsEditing(false);
+    }
+  };
+
+  const handleKeyUp = (e: React.KeyboardEvent) => {
+    if (e.key === 'Enter') {
+      handleEditSubmit();
+    } else if (e.key === 'Escape') {
+      setIsEditing(false);
+    }
+  };
+
+  const handleTitleBlur = () => {
+    handleEditSubmit();
+  };
+
+  return (
+    <div
+      data-cy="Todo"
+      className={classNames('todo', { completed: todo.completed })}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          onChange={() => activeTodo(todo.id, todo.completed)}
+        />
+      </label>
+
+      {!isEditing ? (
+        <>
+          <span
+            data-cy="TodoTitle"
+            className="todo__title"
+            onDoubleClick={() => setIsEditing(true)}
+          >
+            {todoTitle}
+          </span>
+          <button
+            type="button"
+            className="todo__remove"
+            data-cy="TodoDelete"
+            onClick={() => !isLoading && deleteTodo(todo.id)}
+          >
+            ×
+          </button>
+        </>
+      ) : (
+        <form
+          onSubmit={(e) => {
+            e.preventDefault();
+            handleEditSubmit();
+          }}
```

Comment:
move  `e.preventDefault();` into `handleEditSubmit` handler
```suggestion
          onSubmit={handleEditSubmit}
```